File: ios/Helix/Features/Onboarding/OnboardingData.swift
/**
 * Onboarding Data Model - Helix iOS
 * Shared state across all onboarding steps
 */

import Foundation

struct OnboardingData {
    var instanceKey: String = UUID().uuidString
    var keySaved: Bool = false
    var desktopInstructionsViewed: Bool = false
    var gatewayConnected: Bool = false
    var gatewayUrl: String = "wss://gateway.helix-project.org"
    var connectionError: String?
}

extension OnboardingData {
    mutating func reset() {
        self = OnboardingData()
    }

    var isStep1Complete: Bool {
        !instanceKey.isEmpty
    }

    var isStep2Complete: Bool {
        keySaved
    }

    var isStep3Complete: Bool {
        desktopInstructionsViewed
    }

    var isStep4Complete: Bool {
        gatewayConnected && connectionError == nil
    }
}

File: android/app/src/main/java/com/helix/features/onboarding/models/OnboardingData.kt
/**
 * Onboarding Data Model - Helix Android
 * Shared state across all onboarding steps
 */

package com.helix.features.onboarding.models

data class OnboardingData(
    val instanceKey: String = java.util.UUID.randomUUID().toString(),
    val keySaved: Boolean = false,
    val desktopInstructionsViewed: Boolean = false,
    val gatewayConnected: Boolean = false,
    val gatewayUrl: String = "wss://gateway.helix-project.org",
    val connectionError: String? = null
) {
    val isStep1Complete: Boolean get() = instanceKey.isNotEmpty()
    val isStep2Complete: Boolean get() = keySaved
    val isStep3Complete: Boolean get() = desktopInstructionsViewed
    val isStep4Complete: Boolean get() = gatewayConnected && connectionError == null

    fun reset(): OnboardingData = OnboardingData()

    fun copy(
        instanceKey: String = this.instanceKey,
        keySaved: Boolean = this.keySaved,
        desktopInstructionsViewed: Boolean = this.desktopInstructionsViewed,
        gatewayConnected: Boolean = this.gatewayConnected,
        gatewayUrl: String = this.gatewayUrl,
        connectionError: String? = this.connectionError
    ) = OnboardingData(
        instanceKey = instanceKey,
        keySaved = keySaved,
        desktopInstructionsViewed = desktopInstructionsViewed,
        gatewayConnected = gatewayConnected,
        gatewayUrl = gatewayUrl,
        connectionError = connectionError
    )
}
